src/OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteDynamics.cpp

```diff
@@ -0,0 +1,348 @@
+{
+
+    aSatelliteDynamics.print(anOutputStream) ;
+
+    return anOutputStream ;
+
+}
+
+bool                            SatelliteDynamics::isDefined                ( ) const
+{
+
+    bool stateVectorDimensionIsDefined ;
+    switch (stateVectorDimension_)
+    {
+        case SatelliteDynamics::StateVectorDimension::PositionVelocity:
+            stateVectorDimensionIsDefined = true ;
+            break ;
+        case SatelliteDynamics::StateVectorDimension::PositionVelocitywithDragCoefficient:
+            stateVectorDimensionIsDefined = true ;
+            break ;
+        default:
+            stateVectorDimensionIsDefined = false ;
+            break ;
+    }
+
+    return environment_.isDefined() && satelliteSystem_.isDefined() && state_.isDefined() && stateVectorDimensionIsDefined ;
+}
+
+void                            SatelliteDynamics::print                    (           std::ostream&               anOutputStream,
+                                                                                        bool                        displayDecorator                            ) const
+{
+
+    displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "SatelliteDynamics") : void () ;
+    ostk::core::utils::Print::Line(anOutputStream) << "State vector type:"                << SatelliteDynamics::StringFromStateVectorDimension(stateVectorDimension_) ;
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Environment:" << environment_ ;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Satellite System") ;
+    satelliteSystem_.print(anOutputStream, false) ;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "State") ;
+    state_.print(anOutputStream, false) ;
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void () ;
+
+}
+
+SatelliteDynamics::StateVectorDimension       SatelliteDynamics::getStateVectorDimension                            ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("SatelliteDynamics") ;
+    }
+
+    return stateVectorDimension_ ;
+
+}
+
+State                           SatelliteDynamics::getState                 ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Satellite Dynamics") ;
+    }
+
+    return state_ ;
+
+}
+
+void                            SatelliteDynamics::setState                 (   const   State&                      aState                                      )
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Satellite Dynamics") ;
+    }
+
+    state_ = aState ;
+    environment_.setInstant(aState.getInstant()) ;
+
+}
+
+SatelliteDynamics::DynamicalEquationFuncCallback  SatelliteDynamics::accessDynamicalEquations             ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("SatelliteDynamics") ;
```

Comment:
This form of dynamical equation may not be the most elegant, but it will have to do for now (with the time I have left) since I can't just getGravitationalFieldAt for the Sun and the Moon since it throws an Undefined error for them when I do.